Python/bytecodes.c

```diff
@@ -0,0 +1,3795 @@
+#include "Python.h"
+
+#include "opcode.h"
+#include "pycore_atomic.h"
+#include "pycore_frame.h"
+
+void _PyFloat_ExactDealloc(PyObject *);
+void _PyUnicode_ExactDealloc(PyObject *);
+
+#define SET_TOP(v)        (stack_pointer[-1] = (v))
+#define GETLOCAL(i)     (frame->localsplus[i])
+
+#define inst(name, stack_effect) case name:
+#define family(name) static int family_##name
+
+static void
+dummy_func(
+    PyThreadState *tstate,
+    _PyInterpreterFrame *frame,
+    unsigned char opcode,
+    unsigned int oparg,
+    _Py_atomic_int * const eval_breaker,
+    _PyCFrame cframe,
+    PyObject *names,
+    PyObject *consts,
+    _Py_CODEUNIT *next_instr,
+    PyObject **stack_pointer
+)
+{
+    switch (opcode) {
+
+// BEGIN BYTECODES //
+        inst(NOP, ( -- )) {
+        }
+
+        inst(RESUME, ( -- )) {
+            _PyCode_Warmup(frame->f_code);
+            JUMP_TO_INSTRUCTION(RESUME_QUICK);
+        }
+
+        inst(RESUME_QUICK, ( -- )) {
+            PREDICTED(RESUME_QUICK);
+            assert(tstate->cframe == &cframe);
+            assert(frame == cframe.current_frame);
+            if (_Py_atomic_load_relaxed_int32(eval_breaker) && oparg < 2) {
+                goto handle_eval_breaker;
+            }
+        }
+
+        inst(LOAD_CLOSURE, ( -- __0)) {
+            /* We keep LOAD_CLOSURE so that the bytecode stays more readable. */
+            PyObject *value = GETLOCAL(oparg);
+            if (value == NULL) {
+                goto unbound_local_error;
+            }
+            Py_INCREF(value);
+            PUSH(value);
+        }
+
+        inst(LOAD_FAST_CHECK, ( -- __0)) {
```

Comment:
I can easily get rid of the stack effect, but my thinking here is that the code generator would ignore "dummy" stack effects that begin with `__` (currently it ignores all stack effects). We could then later use these as a starting point for manual conversion to correct stack effects. It's true that technically the dummy stack effects are redundant because the real stack effect is present in the code, and in some cases the dummy stack effect is incorrect (because it's derived from the `stack_effect()` function, which doesn't distinguish between e.g. `(x -- y, z)` and `( -- z)`).

I had hoped it would help, not hinder with conversion, but if you think it's confusing I'll get rid of it.